ui/hooks/useTokenDetectionPolling.ts

```diff
@@ -0,0 +1,37 @@
+import { useSelector } from 'react-redux';
+import {
+  getAllDetectedTokensForSelectedAddress,
+  getNetworkConfigurationsByChainId,
+  getUseTokenDetection,
+} from '../selectors';
+import {
+  tokenDetectionStartPolling,
+  tokenDetectionStopPollingByPollingToken,
+} from '../store/actions';
+import useMultiPolling from './useMultiPolling';
+
+const useTokenDetectionPolling = () => {
+  const useTokenDetection = useSelector(getUseTokenDetection);
+  const networkConfigurations = useSelector(getNetworkConfigurationsByChainId);
+
+  const chainIds = Object.keys(networkConfigurations);
+
+  // Selectors returning state updated by the polling
+  const detectedTokensAllChains = useSelector(
+    getAllDetectedTokensForSelectedAddress,
+  );
+
+  console.log('detectedTokensAllChains *****', detectedTokensAllChains);
+
+  useMultiPolling({
+    startPolling: tokenDetectionStartPolling,
+    stopPollingByPollingToken: tokenDetectionStopPollingByPollingToken,
+    input: useTokenDetection ? chainIds : [],
+  });
+
+  return {
+    detectedTokensAllChains,
```

Comment:
the idea here is to return list of detected tokens accross all chains mapped by chainId, example:
```
{ '0x1': { detected token ...} , ....}
```